lib/Varien/Data/Collection/Db.php

```diff
@@ -775,7 +786,7 @@ protected function _saveCache($data, $select)
     /**
      * Check if cache can be used for collection data
      *
-     * @return bool
+     * @return Zend_Cache_Core
```

Comment:
I would say this should stay bool, as everywhere in the core it's used as bool.
In a separate patch we can change the return type to bool always.

Reply:
> I would say this should stay bool, as everywhere in the core it's used as bool.
> In a separate patch we can change the return type to bool always.

Then everywhere else it is wrong too. Again, the "Doc-Updates" depend on each other ... i'm pretty sure it has been corrected in all classes (also Varien_). Espeacially "core methods" like this have been checked manually via xdebug.

This sh*** took probably the most time ... :(

Reply:
@tmotyl it is `Zend_Cache_Core|false`.